sklearn/tree/_tree.pyx

```diff
@@ -95,48 +96,129 @@ NODE_DTYPE = np.dtype({
 # =============================================================================
 
 cdef class Criterion:
-    """Interface for impurity criteria."""
+    """Interface for impurity criteria. 
+
+    This object stores methods on how to calculate how good a split is using 
+    different metrics.
+    """
 
     cdef void init(self, DOUBLE_t* y, SIZE_t y_stride, DOUBLE_t* sample_weight,
                    double weighted_n_samples, SIZE_t* samples, SIZE_t start,
                    SIZE_t end) nogil:
-        """Initialize the criterion at node samples[start:end] and
-           children samples[start:start] and samples[start:end]."""
+        """Placeholder for a method which will initialize the criterion.
+
+        Parameters
+        ----------
+        y: array-like, dtype=DOUBLE_t
+            This is a vector of targets in a memory-efficient buffer format, 
+            used to evaluate the split
+        y_stride: SIZE_t
+            This indicates the stride of the buffer to allow access to elements
+            easily
+        sample_weight: array-like, dtype=DOUBLE_t
+            The weight of each sample
+        weighted_n_samples: DOUBLE_t
+            The total weight of the samples being considered
+        samples: array-like, dtype=DOUBLE_t
+            The input data which is being split
+        start: SIZE_t
+            The first sample to be used on this leaf
+        end: SIZE_t
+            The last sample used in this leaf
+
+        """
+        
         pass
 
     cdef void reset(self) nogil:
-        """Reset the criterion at pos=start."""
+        """Placeholder for a method which will reset the criterion at 
+        pos=start.
+        """
+        
         pass
 
     cdef void update(self, SIZE_t new_pos) nogil:
-        """Update the collected statistics by moving samples[pos:new_pos] from
-           the right child to the left child."""
+        """
+        This is a placeholder for a method which will update the collected 
+        statistics by moving samples[pos:new_pos] from the right child to the
+        left child.
+
+        Parameters
+        ----------
+        new_pos: SIZE_t
+            The index of the final sample to be moved over.
```

Comment:
This would be more informative to say :
``New starting index of the samples in the right child.``